app/actors/sufia/create_with_files_actor.rb

```diff
@@ -3,7 +3,7 @@ module Sufia
   class CreateWithFilesActor < CurationConcerns::Actors::AbstractActor
     def create(attributes)
       self.uploaded_file_ids = attributes.delete(:uploaded_files)
-      validate_files && mark_inactive && next_actor.create(attributes) && attach_files
+      validate_files && next_actor.create(attributes) && attach_files && mark_inactive
```

Comment:
This should be a new actor, it should not go into this class. Add your actor to the stack here: https://github.com/projecthydra/sufia/blob/master/app/services/sufia/actor_factory.rb#L3